pkg/test/harness.go

```diff
@@ -218,23 +199,40 @@ func (h *Harness) Run() {
 
 	defer h.Stop()
 
-	if h.StartKUDO || h.StartControlPlane {
-		if err := h.RunKUDO("../../config/crds/"); err != nil {
-			h.T.Fatal(err)
-		}
+	cl, err := h.Client(false)
+	if err != nil {
+		h.T.Fatal(err)
 	}
 
-	cl, err := h.Client()
+	dClient, err := h.DiscoveryClient()
 	if err != nil {
 		h.T.Fatal(err)
 	}
 
-	dClient, err := h.DiscoveryClient()
+	// Install CRDs
+	crds, err := testutils.InstallManifests(context.TODO(), cl, dClient, h.CRDDir)
 	if err != nil {
 		h.T.Fatal(err)
 	}
 
-	if err := testutils.InstallManifests(context.TODO(), cl, dClient, h.ManifestsDir); err != nil {
+	if err := testutils.WaitForCRDs(dClient, crds); err != nil {
+		h.T.Fatal(err)
+	}
+
+	// Create a new client to bust the client's CRD cache.
+	cl, err = h.Client(true)
+	if err != nil {
+		h.T.Fatal(err)
+	}
+
+	if h.StartKUDO || h.StartControlPlane {
+		if err := h.RunKUDO(); err != nil {
+			h.T.Fatal(err)
+		}
+	}
+
+	// Install required manifests.
+	if _, err := testutils.InstallManifests(context.TODO(), cl, dClient, h.ManifestsDir); err != nil {
```

Comment:
so these manifests are the framework manifests? Could we do less generic name? I mean I feel like we're way past being a generic kubernetes testing framework since we have fields mentioning kudo directly

Reply:
Also we don't need to await here?

Reply:
These are any manifests that the user provides and not limited to frameworks. I have some thoughts on how we’ll install frameworks more cleanly going forward (I’d like to have it fetch them from the repository and install them that way - ideally calling or re-using the kudoctl install code). So I think it makes more sense to call it manifests, since it can be anything.